File: Recursion/count_digits.cpp
// Count the number of digits in an integer
// Sample Input: 2003
// Output: 4

#include<bits/stdc++.h>
using namespace std;
int count_digits(int n){
    if(n == 0)
        return 0;       
    int small_no = count_digits(n / 10);
    return small_no + 1;
}
int main(){
    cout << count_digits(2003099910);
    return 0;
}

File: Recursion/count_digits.go
// Count the number of digits in an integer
package main

import "fmt"

func CountDigits(n int) int {
	if n == 0 {
		return 0;
	}
	smallNo := CountDigits(n / 10)
	return smallNo + 1
}

func main() {
	msg := CountDigits(2003)
	fmt.Println(msg)
}